create_pdfs.py

```diff
@@ -97,31 +122,39 @@ def main():
             name_list = NAMES
 
         for name in name_list:
-            cert = CertificateGen(
-                course,
-                args.template_file,
-                aws_id=settings.CERT_AWS_ID,
-                aws_key=settings.CERT_AWS_KEY,
-                dir_prefix=pdf_dir,
-                long_org=args.long_org,
-                long_course=args.long_course,
-                issued_date=args.issued_date,
-            )
-            title = args.assign_title or None
+            if not CertificateGen.reuse_cert(cert, course, designation):
+                cert = CertificateGen(
+                    course,
+                    aws_id=settings.CERT_AWS_ID,
+                    aws_key=settings.CERT_AWS_KEY,
+                    designation=designation,
+                    dir_prefix=pdf_dir,
+                    issued_date=args.issued_date,
+                    long_course=long_course,
+                    long_org=args.long_org,
+                )
             grade = None
             if args.grade_text:
                 grade = args.grade_text
-            (download_uuid, verify_uuid,
-                download_url) = cert.create_and_upload(name, upload=upload_files, copy_to_webroot=False,
-                                                       cleanup=False, designation=title, grade=grade)
+            (download_uuid, verify_uuid, download_url) = cert.create_and_upload(
+                name,
+                cleanup=False,
+                copy_to_webroot=False,
+                grade=grade,
+                upload=upload_files,
+            )
             certificate_data.append((name, course, args.long_org, args.long_course, download_url))
             gen_dir = os.path.join(cert.dir_prefix, S3_CERT_PATH, download_uuid)
             # Remove non-ascii chars from filename before saving locally (This is not the production filename)
             name = ''.join([i if ord(i) < 128 else ' ' for i in name])
-            copy_dest = '{copy_dir}/{course}-{name}.pdf'.format(
+            copy_dest = '{copy_dir}/{course}-{name}'.format(
                 copy_dir=copy_dir,
+                course=course.replace("/", "-"),
```

Comment:
it looks like it was moved to make the definition order match the usage order in the string.